sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_export/_base.py

```diff
@@ -62,22 +66,35 @@ class ExportResult(Enum):
 class BaseExporter:
     """Azure Monitor base exporter for OpenTelemetry."""
 
-    def __init__(self, **kwargs: Any) -> None:
+    def __init__(
+        self,
+        *,
+        credential: Optional["TokenCredential"] = None,
+        disable_offline_storage: bool = False,
+        storage_directory: Optional[str] = None,
+        **kwargs: Any
+    ) -> None:
         """Azure Monitor base exporter for OpenTelemetry.
 
         :keyword str api_version: The service API version used. Defaults to latest.
-        :keyword str connection_string: The connection string used for your Application Insights resource.
-        :keyword ManagedIdentityCredential/ClientSecretCredential credential: Token credential, such as ManagedIdentityCredential or ClientSecretCredential, used for Azure Active Directory (AAD) authentication. Defaults to None.
-        :keyword bool disable_offline_storage: Determines whether to disable storing failed telemetry records for retry. Defaults to `False`.
-        :keyword str storage_directory: Storage path in which to store retry files. Defaults to `<tempfile.gettempdir()>/opentelemetry-python-<your-instrumentation-key>`.
+        :keyword credential: Token credential, such as ManagedIdentityCredential or ClientSecretCredential,
+         used for Azure Active Directory (AAD) authentication. Defaults to None.
+        :paramtype credential: ~azure.core.credentials.TokenCredential
+        :keyword bool disable_offline_storage: Determines whether to disable storing failed telemetry records
+         for retry. Defaults to `False`.
+        :keyword str storage_directory: Storage path in which to store retry files. Defaults
+         to `<tempfile.gettempdir()>/opentelemetry-python-<your-instrumentation-key>`.
         :rtype: None
         """
-        parsed_connection_string = ConnectionStringParser(kwargs.get('connection_string'))
+        parsed_connection_string = ConnectionStringParser(
+            kwargs.get('_conn_str'),
```

Comment:
This is a pretty major change. We would have to update all our our documentation and conflict with the other languages. Why is this important?

Reply:
Not going to mark every spot where this is used for the sake of keeping things clear.

Reply:
We should only support the passing of connection string via one way.
The intention of the addition of the `from_connection_string` classmethod, is that this would become the only way to do so.
Normally I would advocate for a more complete refactor in that parsing of the connection string would happen in the classmethod itself, not the main constructor, however given the complexities of your parser that combines with the env var parsing, I decided the simplest approach for now was just to make the constructor param private/internal-only.

Reply:
I don't understand. We only added from_connection_string because azure sdk previously said to. If we removed it, does that mean we could still have connection string as an argument in the exporter constructor?